cmd/zoekt-dynamic-indexserver/main.go

```diff
@@ -132,6 +137,21 @@ func (s *indexServer) serveHealthCheck(w http.ResponseWriter, r *http.Request) {
 	// Nothing to do. Just return 200
 }
 
+var (
+	promRegistry        = prometheus.NewRegistry()
```

Comment:
given this is attached to the indexServer, why not just make it part of indexServer rather than a global? Alternatively just use the DefaultRegistry if you want to rely on globals.

Either way you mix `init()` time registring on this with runtime. That is a path towards shooting yourself in the foot down the road.

Reply:
@keegancsmith Thank you for the suggestion! I've extracted initialization code into `func (s *indexServer) initMetrics()`.